src/neo4j_genai/experimental/components/entity_relation_extractor.py

```diff
@@ -139,30 +143,42 @@ def __init__(
         create_lexical_graph: bool = True,
         **kwargs: Any,
     ) -> None:
-        self.create_lexical_graph = create_lexical_graph
         self.on_error = on_error
-        self._id_prefix = ""
+        self.create_lexical_graph = create_lexical_graph
 
     @abc.abstractmethod
-    async def run(self, chunks: TextChunks, **kwargs: Any) -> Neo4jGraph:
+    async def run(
+        self,
+        chunks: TextChunks,
+        document_info: Optional[DocumentInfo] = None,
+        **kwargs: Any,
+    ) -> Neo4jGraph:
         pass
 
-    def update_ids(self, graph: Neo4jGraph, chunk_index: int) -> Neo4jGraph:
+    def update_ids(
+        self, graph: Neo4jGraph, chunk_index: int, run_id: str
+    ) -> Neo4jGraph:
         """Make node IDs unique across chunks and pipeline runs by
         prefixing them with a custom prefix (set in the run method)
         and chunk index."""
+        prefix = f"{run_id}:{chunk_index}"
         for node in graph.nodes:
-            node.id = f"{self._id_prefix}:{chunk_index}:{node.id}"
+            node.id = f"{prefix}:{node.id}"
             if node.properties is None:
                 node.properties = {}
             node.properties.update({"chunk_index": chunk_index})
         for rel in graph.relationships:
-            rel.start_node_id = f"{self._id_prefix}:{chunk_index}:{rel.start_node_id}"
-            rel.end_node_id = f"{self._id_prefix}:{chunk_index}:{rel.end_node_id}"
+            rel.start_node_id = f"{prefix}:{rel.start_node_id}"
+            rel.end_node_id = f"{prefix}:{rel.end_node_id}"
         return graph
 
+
+class LexicalGraphBuilder:
```

Comment:
Why create a class for this if it's mostly just a collection of useful methods?

Reply:
Yes, what do you suggest? It used to be in the `EntityRelationExtractor` base class but it was forcing us to instantiate an abstract class to test it.

Reply:
You can't run the unit tests on these methods without moving them to a separate class?